### PR TITLE
Some maintenance & clean-up

### PR DESCRIPTION
**Pull request by niknetniko**
- Remove thesis text
- Add more docs for dependencies
- Remove duplicate docker image (there is one in the docker images repo)
- Remove the use of env variables for linters:
   - checkstyle should be available as a system package on most OS
   - ktlint is available on some, but otherwise an executable jar can be added to the path
- Update versions of a bunch of stuff.

TODO:
- [ ] Update & test docker image.

Obsoletes parts of #282, by going to 1.6 directly. The language support still needs to be adjusted.

### CODE REVIEW

File: dependencies.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+| Name                    | Versions | Installation |
+|-------------------------|----------|--------------|
+| `eslint`                | 8.7      | npm package  |
+| `abstract-syntax-tree`  | 2.16     | npm package  |
+
+Install npm packages as follows:
+
+```shell
+$ npm install eslint@7 abstract-syntax-tree@2.16
```
Comment (bsels) [inline dependencies.md:89]: Typo in version eslint.
```suggestion
$ npm install eslint@8.7 abstract-syntax-tree@2.16
```

File: tested/languages/java/templates/value_basic.mako [hunk 2]
```diff
@@ -4,7 +4,12 @@
 <%! from json import dumps %>\
 <%page args="value" />\
 % if value.type == BasicNumericTypes.INTEGER:
-    ${value.data}\
+    ## Basic heuristic for long numbers
+    % if len(str(value.data)) > 9:
```
Comment (bsels) [inline tested/languages/java/templates/value_basic.mako:8]: Can you just check for the actual number?
```suggestion
    % if value.data > (2**31 - 1):
```

Comment (niknetniko) [reply tested/languages/java/templates/value_basic.mako:8]: Good point actually!

File: tested/languages/java/templates/value_basic.mako [hunk 3]
```diff
@@ -4,7 +4,12 @@
 <%! from json import dumps %>\
 <%page args="value" />\
 % if value.type == BasicNumericTypes.INTEGER:
-    ${value.data}\
+    ## Basic heuristic for long numbers
+    % if value.data > (2**31 - 1):
```
Comment (bsels) [inline tested/languages/java/templates/value_basic.mako:8]: We have forgotten the check for negative values
```suggestion
    % if (value.data > (2**31 - 1)) or (value.data < 2**31):
```

**Pull request merged**